contributing.md

```diff
@@ -0,0 +1,163 @@
+* Promote beets! Help get the word out by telling your friends, writing a blog
+  post, or discussing it on a forum you frequent.
+* Improve the [documentation][docs]. It's incredibly easy to contribute here:
+  just find a page you want to modify and hit the "Edit on GitHub" button in
+  the upper-right. You can automatically send us a pull request for your
+  changes.
+* GUI design. For the time being, beets is a command-line-only affair. But
+  that's mostly because I don't have any great ideas for what a good GUI
+  should look like. If you have those great ideas, please get in touch.
+* Benchmarks. We'd like to have a consistent way of measuring speed
+  improvements in beets' tagger and other functionality as well as a way of
+  comparing beets' performance to other tools. You can help by compiling a
+  library of freely-licensed music files (preferably with incorrect metadata)
+  for testing and measurement.
+* Think you have a nice config or unique use-case for beets? We'd love to hear about it! Submit a post to our [our forums](https://discourse.beets.io/) under the label 'recipe' for a chance to get featured in our [docs](https://beets.readthedocs.io/en/stable/guides/advanced.html).
+* Consider helping out in [our forums](https://discourse.beets.io/) by responding to support requests or driving some new discussions. 
+
+[docs]: http://beets.readthedocs.org/
+
+## Programming contributions
+* As a programmer (even if you're just a beginner!), you have a ton of opportunities to get your feet wet with `beets`. 
+* For developing plugins, or hacking away at beets, there's some good information in the [for developers](https://beets.readthedocs.io/en/stable/dev/) section of the docs.
+
+### Getting the Source
+
+The easiest way to get started with the latest beets source is to use [pip](http://pip.openplans.org/)  to install an "editable" package. This can be done with one command:
+
+    $ pip install -e git+https://github.com/beetbox/beets.git#egg=beets
+
+Or, equivalently:
+
+    $ git clone https://github.com/beetbox/beets.git
+    $ cd beets
+    $ pip install -e .
+
+If you already have a released version of beets installed, you may need to
+remove it first by typing `pip uninstall beets`. The pip command above will put
+the beets source in a `src/beets` directory and install the `beet` CLI script to
+a standard location on your system. You may want to use the `--src` option to specify
+the parent directory where the source will be checked out and the `--user` option
+such that the package will be installed to your home directory (compare with the output of
+`pip install --help`). 
+
+Alternatively, you can get the source via 
+[git at GitHub](http://github.com/beetbox/beets). The pip method above installs
+dependencies automatically, but if you do it manually, you can use pip to
+install these Python modules:  
+* `mutagen`
+* `munkres`
+* `unidecode`
+* `pyYAML`
+* `musicbrainzngs`
+* `confuse`
+* `mediafile`
+* `jellyfish`
```

Comment:
Maybe let's just get rid of this section—I think `pip install -e` should probably be the recommendation for basically everybody.